src/System.Runtime.WindowsRuntime/src/System/IO/StreamOperationAsyncResult.cs

```diff
@@ -150,7 +150,8 @@ internal bool CancelStreamOperation()
             if (_callbackInvoked)
                 return false;
 
-            _asyncStreamOperation.Cancel();
+            _asyncStreamOperation?.Cancel();
+            _asyncStreamOperation = null;
```

Comment:
Dan's feedback and my feedback conflicted in the resulting code.  There's no value in setting _asyncStreamOperation to null if it's already null.  If we weren't doing that, then it makes sense to do:
```C#
_asyncStreamOperation?.Cancel();
```
instead of the exact equivalent:
```C#
if (_asyncStreamOperation != null)
{
    _asyncStreamOperation.Cancel();
}
```
but since you're also setting the field to null, it makes sense to do:
```C#
if (_asyncStreamOperation != null)
{
    _asyncStreamOperation.Cancel();
    _asyncStreamOperation = null;
}
```
instead of what's currently there, as then we're using the existing null check to avoid setting the field to null when it already is.